library/src/main/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapter.java

```diff
@@ -290,7 +320,12 @@ public int getItemViewType(int position) {
 
                 switch (section.getState()) {
                     case LOADED:
-                        return viewType + 2;
+                        int itemViewType = section.getItemViewType(getPositionInSection(position));
+                        if (itemViewType == -1) {
+                            return viewType + 2;
+                        } else {
+                            return itemViewType;
```

Comment:
[CODE] Unit tests should be created/changed accordingly to test the new logic/output.